internal/agent/agent_test.go

```diff
@@ -951,3 +951,53 @@ func TestContext_SendEvent_LogTruncatedEvent(t *testing.T) {
 	assert.Contains(t, written, fmt.Sprintf("original=\"+map[key:%s]", original))
 	assert.Contains(t, written, fmt.Sprintf("truncated=\"+map[key:%s]", truncated))
 }
+
+func TestRunsWithCloudProvider(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name          string
+		cloudProvider string
+		assertFunc    func(assert.TestingT, error, ...any) bool
+		retries       int
+	}{
+		{
+			name:          "Valid cloud tries (and fails) to get metadata",
+			cloudProvider: "aws",
+			assertFunc:    assert.Error,
+			retries:       0,
+		},
+		{
+			name:          "Valid cloud tries (and fails) to get metadata after 3 retries",
+			cloudProvider: "aws",
+			assertFunc:    assert.Error,
+			retries:       3,
+		},
+	}
+
+	for _, tt := range tests {
+		testCase := tt
+
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+
+			waitGrp := sync.WaitGroup{}
+			waitGrp.Add(1)
+
+			//nolint:exhaustruct
+			agt := newTesting(&config.Config{
+				CloudProvider:      testCase.cloudProvider,
+				CloudMaxRetryCount: testCase.retries,
+			})
+
+			var err error
+			go func() {
+				err = agt.Run()
```

Comment:
Is this wait group necessary here? It's waiting for `agent.Run` to finish so I think that it can be replaced with `agt.Run` with no goroutine
```go
		t.Run(testCase.name, func(t *testing.T) {
			t.Parallel()

			//nolint:exhaustruct
			agt := newTesting(&config.Config{
				CloudProvider:      testCase.cloudProvider,
				CloudMaxRetryCount: testCase.retries,
			})

			err := agt.Run()
			testCase.assertFunc(t, err)
		})
```

Reply:
Yeah, makes sense!